### DISCUSSION TITLE
Rancher Desktop Linux ARM64 support

### DISCUSSION THREAD

**Discussion by Anutrix** [rancher-sandbox/rancher-desktop#2283] [Q&A]
Rancher Desktop added support for Apple Silicon but what about Linux ARM64 support? Is it already supported?
I could only find [rancher-desktop-linux-v1.3.0.zip](https://github.com/rancher-sandbox/rancher-desktop/releases/download/v1.3.0/rancher-desktop-linux-v1.3.0.zip) but it seems like x86_64.

There's no documentation or note regarding this. aarch64 mentions I could find in the repo were specifically for Apple Silicon.
Any info would be appreciated.

**Comment by jandubois** [comment]
> what about Linux ARM64 support?

This is mostly because there has been no demand for it; I believe you are the first person asking about it.

I think it would be comparatively easy to do, but since it will add extra work for each release (there is a certain amount of manual testing we do in addition to CI), so we don't want to waste the effort, if nobody is using it.

Also none of the developers are using a Linux on arm workstation, adding to the effort required.

Did you try building it yourself? Did you run into any problems?

**Reply by Anutrix** [reply]
I don't have real ARM64 Linux device but have some Oracle Cloud VMs running Ubuntu ARM64. I will test on those when I have time.

**Comment by kcq** [comment]
@jandubois or anybody else on the Desktop side... wonder if there's been any changes since the original discussion. Linux ARM64/AArch64 laptops are more common now (early 2024) even though they are not mainstream yet. Asahi Linux also makes it possible to have ARM64-based Linux (they now have a very nice Fedora installer :-)). How likely is it for the Rancher team to have an AppImage or other type of installer for this architecture? Is DIY still the best bet?

**Reply by jandubois** [reply]
I'm planning to make sure all the prerequisites for Rancher Desktop will be built for Linux and Windows on ARM64, e.g. I just added the Windows on ARM64 targets to docker-cli: https://github.com/rancher-sandbox/rancher-desktop-docker-cli/pull/9/files

I will add the others over the coming weeks (or file PRs for upstream assets). Once all the pieces are in place we can update the postinstall script to pull the right binaries for ARM64 builds, to make it a lot easier for people to build from source.

There are still no plans currently for SUSE to publish Linux or Windows ARM64 binaries; mostly due to the additional QA effort (and hardware requirements) it would add to each release. As I mentioned before, demand still seems to be very low for these platforms.

But we can discuss this again once ARM64 builds are done automatically via CI (there are no Github runners for Windows/Linux on ARM64, so we can't even run the automated tests in CI).

**Reply by kcq** [reply]
Sounds pretty reasonable. Yes, this is not the top architecture to support. Building from source is the next best option. 

Actuated ( https://actuated.dev/ ) is a nice option to do Arm Github Action builds. They have a pilot program for CNCF projects to do free builds, but Rancher Desktop is not in CNCF though :)

**Comment by ninlil** [comment]
Just got my Surface Pro 11 with an ARM CPU, and would very much like to install Rancher Desktop for all my local container management (both as plain containers and in the local Kubernetes)

It's now 1+ year since the last comment here... is this coming anytime at all, or is the other "Xxxxxx Desktop"-product the only way to go?

**Comment by zoosky** [comment]
Same here, on Fedora Asahi Linux
